src/core/thread/mle_router.cpp

```diff
@@ -4364,19 +4364,31 @@ otError MleRouter::AppendChildAddresses(Message &aMessage, Child &aChild)
             entry.SetUncompressed();
             entry.SetIp6Address(address);
         }
-        else
+        else if (context.mContextId != kRsvContextId)
         {
             // compressed entry
             entry.SetContextId(context.mContextId);
             entry.SetIid(address.GetIid());
         }
+        else
+        {
+            continue;
```

Comment:
Theoretical considerations.. I think we had this discussion in some other PR some time ago without any conclusion.

Is it valid to create (statically) and then register extra address using Mesh Local Prefix (other than RLOC/ALOC/ML-EID) according to Thread Specification e.g. `MLP::1`? Indirectly, because of the sentence, you wrote in PR description, we can deduce that it is not. Since a parent, can't distinguish which address is ML-EID and which one is an extra address. Or maybe we can treat just the first address with Context ID = 0 as ML-EID.

I believe that there is one Zephyr example of OpenThread that consist of two applications (server and client). In order to make the example simple, without implementing a protocol like mDNS to discover the address of the server, the client uses hardcoded address `MLP::1` (client has address `MLP::2`). OpenThread allows for that, but is that a valid simplification then? The use case is a little bit different since two devices are FTD and so there is no explicit address registration there but... (@rlubos - FYI)

Reply:
Spec only says the thread interface autoconfigures RLOC(/ALOC) and ML-EID, and no clear text about whether to allow extra MLP address by user.
However for my personal understanding, Thread may have the assumption that there would be only one ML-EID. 
@jwhui do you have any inputs?
Maybe we can file a JIRA to Thread Group to discuss this?

Reply:
I don't think the Thread Spec explicitly precludes assignment of additional IPv6 addresses using the Mesh Local Prefix. At the same time, the Thread Spec does not provide any mechanism for a parent to distinguish between the ML-EID from other IPv6 addresses with the Mesh Local Prefix (as @LuDuda points out above). For that reason, I don't think Thread currently allows for additional IPv6 addresses with the Mesh-Local Prefix.